connectors/tests/test_sync.py

```diff
@@ -463,6 +462,7 @@ def jobs_update(url, **kw):
     )
     await service.run()  # fails
     await service.run()  # works
```

Comment:
There are two tests (`test_connector_service_poll_clear_error` and `test_connector_settings_change`) calling `service.run()` twice. Is it expected to have the service run multiple times? I expect the 2nd call should fail with some message that the service is already running. cc. @elastic/ingestion-team